packages/liquidation-sdk-viem/src/preLiquidation/logGetters.ts

```diff
@@ -0,0 +1,207 @@
+import { type Address, ChainUtils, type MarketId } from "@morpho-org/blue-sdk";
+import { addresses } from "@morpho-org/blue-sdk";
+import type { Account, Chain, Client, Transport } from "viem";
+import { getBlockNumber, getLogs } from "viem/actions";
+import { preLiquidationFactoryConfigs } from "../addresses";
+import type { PreLiquidation } from "./types";
+
+export async function preLiquidationLogs<chain extends Chain = Chain>(
+  client: Client<Transport, chain, Account>,
+): Promise<PreLiquidation[]> {
+  const chainId = ChainUtils.parseSupportedChainId(client.chain.id);
+
+  try {
+    const head = await getBlockNumber(client);
+    const intervals = sliceBlockInterval(
+      preLiquidationFactoryConfigs[chainId].startBlock,
+      BigInt(head),
+    );
+
+    const logs = (
+      await Promise.all(
+        intervals.map((interval) =>
+          getLogs(client, {
```

Comment:
I think it'd be cleaner to create a viem contract object and call `getContractEvents(...)`. Inlining the ABI doesn't feel right.